pkg/syncer/cnsoperator/controller/cnsregistervolume/cnsregistervolume_controller.go

```diff
@@ -557,53 +558,65 @@ func (r *ReconcileCnsRegisterVolume) Reconcile(ctx context.Context,
-		if apierrors.IsAlreadyExists(err) {
-			log.Infof("PVC: %s already exists", instance.Spec.PvcName)
-			pvc, err = k8sclient.CoreV1().PersistentVolumeClaims(instance.Namespace).Get(ctx,
-				instance.Spec.PvcName, metav1.GetOptions{})
+
+	if pvc != nil {
+		log.Infof("PVC: %s already exists", instance.Spec.PvcName)
+		if pvc.Status.Phase == v1.ClaimBound && pvc.Spec.VolumeName != pvName {
+			// This is handle cases where PVC with this name already exists and
+			// is bound. This happens when a new CnsRegisterVolume instance is
+			// created to import a new volume with PVC name which is already
+			// created and is bound.
+			msg := fmt.Sprintf("Another PVC: %s already exists in namespace: %s which is Bound to a different PV",
+				instance.Spec.PvcName, instance.Namespace)
+			log.Errorf(msg)
+			setInstanceError(ctx, r, instance, msg)
+			// Untag the CNS volume which was created previously.
+			_, err = common.DeleteVolumeUtil(ctx, r.volumeManager, volumeID, false)
 			if err != nil {
-				msg := fmt.Sprintf("Failed to get PVC: %s on namespace: %s", instance.Spec.PvcName, instance.Namespace)
-				log.Errorf(msg)
-				setInstanceError(ctx, r, instance, msg)
-				return reconcile.Result{RequeueAfter: timeout}, nil
-			}
-			if pvc.Status.Phase == v1.ClaimBound && pvc.Spec.VolumeName != pvName {
-				// This is handle cases where PVC with this name already exists and
-				// is bound. This happens when a new CnsRegisterVolume instance is
-				// created to import a new volume with PVC name which is already
-				// created and is bound.
-				msg := fmt.Sprintf("Another PVC: %s already exists in namespace: %s which is Bound to a different PV",
-					instance.Spec.PvcName, instance.Namespace)
-				log.Errorf(msg)
-				setInstanceError(ctx, r, instance, msg)
-				// Untag the CNS volume which was created previously.
-				_, err = common.DeleteVolumeUtil(ctx, r.volumeManager, volumeID, false)
+				log.Errorf("Failed to untag CNS volume: %s with error: %+v", volumeID, err)
+			} else {
+				// Delete PV created above.
+				err = k8sclient.CoreV1().PersistentVolumes().Delete(ctx, pvName, *metav1.NewDeleteOptions(0))
 				if err != nil {
-					log.Errorf("Failed to untag CNS volume: %s with error: %+v", volumeID, err)
-				} else {
-					// Delete PV created above.
-					err = k8sclient.CoreV1().PersistentVolumes().Delete(ctx, pvName, *metav1.NewDeleteOptions(0))
-					if err != nil {
-						log.Errorf("Failed to delete PV: %s with error: %+v", pvName, err)
-					}
+					log.Errorf("Failed to delete PV: %s with error: %+v", pvName, err)
 				}
-				return reconcile.Result{RequeueAfter: timeout}, nil
 			}
-		} else {
+			return reconcile.Result{RequeueAfter: timeout}, nil
+		}
+
+		if pvc.Spec.DataSourceRef != nil {
+			apiGroup := ""
+			if pvc.Spec.DataSourceRef.APIGroup != nil {
+				apiGroup = *pvc.Spec.DataSourceRef.APIGroup
+			}
+			log.Infof("PVC %s in namespace %s has valid DataSourceRef with apiGroup: %s, kind: %s, name: %s",
```

Comment:
Where do we check *pvc.Spec.DataSourceRef.APIGroup is valid?
If this is already checked within checkExistingPVCDataSourceRef, can we remove it from here?